packages/application-shell-connectors/src/components/application-context/normalizers.ts

```diff
@@ -0,0 +1,218 @@
+ * do `canViewProducts`.
+ *
+ * This function considering its concern belongs into the `permissions` package. However,
+ * for now it doesn't have to be shared and as a result can be co-located with
+ * the fetching logic. Given this mapping needs to be used elsewere feel free
+ * to move this over to `permissions` and export it there.
+ */
+export const normalizeAllAppliedPermissions = (
+  project: TFetchProjectQuery['project']
+) => {
+  if (!project) return null;
+  const allAppliedPermissions = project.allAppliedPermissions;
+  if (allAppliedPermissions.length === 0) {
+    return null;
+  }
+  return allAppliedPermissions.reduce((transformedAllApplied, allApplied) => {
+    if (!allApplied) return transformedAllApplied;
+    return {
+      ...transformedAllApplied,
+      [allApplied.name]: allApplied.value,
+    };
+  }, {});
+};
+
+export const normalizeAllAppliedMenuVisibilities = (
+  project: TFetchProjectQuery['project']
+) => {
+  if (!project) return null;
+  const allAppliedMenuVisibilities = project.allAppliedMenuVisibilities;
+  if (allAppliedMenuVisibilities.length === 0) {
+    return null;
+  }
+  return allAppliedMenuVisibilities.reduce(
+    (transformedAllApplied, allApplied) => {
+      if (!allApplied) return transformedAllApplied;
+      return {
+        ...transformedAllApplied,
+        [allApplied.name]: allApplied.value,
+      };
+    },
+    {}
+  );
+};
+
+type TActionRight = {
+  [key: string]: boolean;
+};
+type NormalizedActionRights = {
+  [key: string]: TActionRight;
+};
+
+export const normalizeAllAppliedActionRights = (
+  project: TFetchProjectQuery['project']
+) => {
+  if (!project) return null;
+  const allAppliedActionRights = project.allAppliedActionRights;
+  if (allAppliedActionRights.length === 0) {
+    return null;
+  }
+  return allAppliedActionRights.reduce<NormalizedActionRights>(
+    (transformedAllApplied, allApplied) => {
+      if (!allApplied) return transformedAllApplied;
+      const previousAllAppliedGroup = transformedAllApplied[allApplied.group];
+      return {
+        ...transformedAllApplied,
+        [allApplied.group]: {
+          ...previousAllAppliedGroup,
+          [allApplied.name]: allApplied.value,
+        },
+      };
+    },
+    {}
+  );
+};
+
+type NormalizedGroupedByPermission = {
+  // E.g. { canManageOrders: { values: [] } }
+  [key: string]: { values: string[] } | null;
+};
+type NormalizedGroupedByResourceType = {
+  // E.g. { orders: {...} }
+  [key: string]: NormalizedGroupedByPermission | null;
+};
+/**
+ * dataFence: {
+ *   store: {
+ *     orders: {
+ *       canManageOrders: { values: ['usa', 'germany'] },
+ *       canViewOrders: { values: ['canada'] },
+ *     }
+ *   }
+ * }
+ */
+type NormalizedStoresGroupByResourceType = {
+  [key: string]: TStoreDataFence[];
+};
+
+const normalizeAppliedDataFencesForStoresByResourceType = (
+  dataFences: Maybe<TStoreDataFence>[]
```

Comment:
Is this a regular `type Maybe<T> = T | null`?

Reply:
It's in the generated types: by default is `export type Maybe<T> = T | null;` but we map it to `export type Maybe<T> = T | undefined;`
See: https://github.com/commercetools/merchant-center-application-kit/blob/bd3eb0d9b2817c1bab08a82e3f558c2853167223/codegen.yml#L47